go.mod

```diff
@@ -6,10 +6,13 @@ require (
 	github.com/k0sproject/k0s v1.27.2-0.20230504131248-94378e521a29
 	github.com/onsi/ginkgo/v2 v2.9.5
 	github.com/onsi/gomega v1.27.7
+	github.com/pkg/errors v0.9.1
```

Comment:
github.com/pkg/errors is legacy https://github.com/pkg/errors/issues/245

 `apierrors.IsNotFound(errors.Cause(err))` should work as `apierrors.IsNotFound(err)`.
`errors.Wrap(..)` => `errors.Join(...)`
`errors.Cause(err) == x`  => `errors.Is(err, x)`

Reply:
Right. I took some inspiration for these from the upstream cluster api providers which seem to use these heavily still. And in the core of cluster api there's lot of usage of `pkg/errors`. I wonder if the changes would be fully compatible 🤔